lib/capistrano/logger.rb

```diff
@@ -102,7 +102,7 @@ def log(level, message, line_prefix=nil)
 
           Logger.sorted_formatters.each do |formatter|
             if (formatter[:level] == level || formatter[:level].nil?)
-              if message =~ formatter[:match] || line_prefix =~ formatter[:match]
+              if message =~ formatter[:match] || formatter[:match] =~ line_prefix.to_s
```

Comment:
line_prefix sometimes is a [Capistrano::ServerDefinition](https://github.com/capistrano/capistrano/blob/v2.15.5/lib/capistrano/server_definition.rb) object which triggers the issue in Ruby 3.2